### PR TITLE
Refine PublicApiAnalyzers multitargeting instructions

### PR DESCRIPTION
**Pull request by jnm2**
/cc @drewnoakes, @mavasani

### CODE REVIEW

File: src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md [hunk 1]
```diff
@@ -35,15 +35,11 @@ For example when using the [`#if` preprocessor directive](https://docs.microsoft
...
-  <ItemGroup Condition="'$(TargetFramework)' == 'net4.8'">
-    <AdditionalFiles Include="net4.8/PublicAPI.Shipped.txt" />
-    <AdditionalFiles Include="net4.8/PublicAPI.Unshipped.txt" />
-  </ItemGroup>
-  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.0'">
-    <AdditionalFiles Include="netcoreapp3.0/PublicAPI.Shipped.txt" />
-    <AdditionalFiles Include="netcoreapp3.0/PublicAPI.Unshipped.txt" />
+  <ItemGroup>
+    <AdditionalFiles Include="PublicAPI/$(TargetFramework)/PublicAPI.Shipped.txt" />
+    <AdditionalFiles Include="PublicAPI/$(TargetFramework)/PublicAPI.Unshipped.txt" />
```
Comment (jnm2) [inline src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md]: The `PublicAPI` folder reduces clutter in the project when you have 3+ target frameworks.

File: src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md [hunk 2]
```diff
@@ -35,15 +35,11 @@ For example when using the [`#if` preprocessor directive](https://docs.microsoft
...
   </ItemGroup>
```
Comment (jnm2) [inline src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md]: 📝 The Properties window in VS 16.6-p2 seems to struggle equally with both this form and the original form, showing only the files associated with the first target framework as AdditionalFiles and the rest as None. PublicApiAnalyzers itself works equally well with both forms.

Comment (mavasani) [reply src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md]: Can you please file an issue in https://github.com/dotnet/project-system repo for the properties window? Thanks for your help!

**Comment by mavasani** [comment]
@drewnoakes Can you please review, and merge if the change looks good to you?

**Comment by jnm2** [comment]
I'm starting to see some odd issues and I may have jumped the gun on this. The analyzer was able to successfully populate each Unshipped file, but it is erroring for the first target framework only as though the Unshipped file for the first target framework was empty.

**Comment by jnm2** [comment]
Yeah, never mind. These problems show up also when using the format originally shown in this document.

**Comment by jnm2** [comment]
Never mind, my problems weren't related to this. Everything is now working file with either project syntax.

File: src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md (hunk 2 above)
Comment (jnm2) [reply src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md]: Sure! https://github.com/dotnet/project-system/issues/6042

File: src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md [hunk 3]
```diff
@@ -35,15 +35,11 @@ For example when using the [`#if` preprocessor directive](https://docs.microsoft
 
 To correctly model the API differences between target frameworks (or any other property), use multiple instances of the `PublicAPI.*.txt` files.
 
-For example, if you target both `net4.8` and `netcoreapp3.0` target frameworks, and APIs differ between each, then you would have the following:
+For example, if you target both `net48` and `netcoreapp3.0` target frameworks, and APIs differ between each, then you would have the following:
```
Comment (drewnoakes) [inline src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md]: I prefer being explicit with the `.` as it is less confusing for people who are new to these monikers. Why not `netcoreapp30` too?

Given the code sample no longer makes explicit mention of any moniker, I'd shorten this sentence to:

```suggestion
If you have multiple target frameworks and APIs differ between them, use the following in your project file:
```

Comment (jnm2) [reply src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md]: > Why not netcoreapp30 too?

I was under the impression that this doesn't work. I think I tried it with netcoreapp1.0 :D

**Pull request merged**